components/feature/push/src/main/java/mozilla/components/feature/push/AutoPushFeature.kt

```diff
@@ -256,6 +264,38 @@ class AutoPushFeature(
         service.deleteToken()
     }
 
+    /**
+     * Verifies status (active, expired) of the push subscriptions and then notifies observers.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun verifyActiveSubscriptions() {
+        DeliveryManager.with(connection) {
+            scope.launchAndTry {
```

Comment:
talked offline: internally we'll expand this method to crash the app on non-transient (e.g. connection) errors, otherwise we're swallowing potential issues

Reply:
Filed: https://github.com/mozilla-mobile/android-components/issues/4824